test/unit/fields_definition_test.rb

```diff
@@ -38,262 +46,228 @@ class FieldsDefinitionTest < ActiveSupport::TestCase
-          @fd.choices_for_views= "changed, values"
-          assert_equal ['changed', 'values'],  @fd.choices
-          assert_equal 'changed, values', @fd.choices_for_views
-        end
-
-        should "work with new line" do
-          @fd.choices_for_views= "Yes, I'm \n No, I'm not"
-          assert_equal ["Yes, I'm", "No, I'm not"], @fd.choices
-          assert_equal "Yes, I'm\nNo, I'm not", @fd.choices_for_views
-        end
-
-        should "work with nil" do
-          @fd.choices_for_views= nil
-          assert_equal [], @fd.choices
-        end
-      end
-    end
+  test "set required, editable and visible properties to required field on target" do
+    field = FieldsDefinition.create(account: @provider, target: 'User', name: 'username', label: 'foo', required: false, hidden: true, read_only: true)
+
+    assert field.required
+    assert_not field.hidden
+    assert_not field.read_only
   end
 
-  context  "not in database" do
-    context "modificable fields" do
-      should "not allow hidden AND required" do
-        fd = FieldsDefinition.new(:account => @provider, :target => 'User',
-                                  :name => 'new_field',
-                                  :hidden => true,
-                                  :label => 'lal',
-                                  :required => true)
-        assert !fd.valid?
-        fd.hidden = false
-        assert fd.valid?
-      end
-
-      should "not allow readonly AND required" do
-        fd = FieldsDefinition.new(:account => @provider, :target => 'User',
-                                  :name => 'new_field',
-                                  :read_only => true,
-                                  :label => 'lal',
-                                  :required => true)
-        assert !fd.valid?
-        fd.read_only = false
-        assert fd.valid?
-      end
-
-      should "allow choices property " do  #if type = string?
-        a = FieldsDefinition.new(:account => @provider, :label => 'lal', :target => 'User', :name => 'first_name', :choices => ["a", "b"])
-        assert a.valid?
-      end
-    end
+  test "builtin fields required field on target allow choices property" do
+    fd = new_field_definition(choices: %w[opt1 opt2])
+    assert fd.valid?
   end
 
-  context '.editable_by' do
-    setup do
-      @provider = FactoryBot.create :provider_account
-      FieldsDefinition.delete_all
-      FactoryBot.create(:fields_definition, :account => @provider, :name => "hidden",
-              :hidden => true)
-      FactoryBot.create(:fields_definition, :account => @provider, :name => "read_only",
-              :read_only => true)
-      @public_field = FactoryBot.create(:fields_definition, :account => @provider,
-                              :name => "public")
-      @buyer = FactoryBot.create :buyer_account, :provider_account => @provider
-    end
+  test "builtin fields and are modifiable fields not allow hidden AND required" do
```

Comment:
I can't parse this test name, do you know what it should mean? :)

Reply:
```suggestion
  test "builtin modifiable fields not allow hidden AND required" do
```

Reply:
Absolutely no idea haha. I think we should probably find a better test name? Like, what's a _modifiable field_ anyway? And what's _builtin_ and what isn't.
I guess we could simply say _field definition_? @hallelujah any ideas/opinions?